pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -579,49 +588,241 @@ func (og *operationGenerator) GenerateUnmountDeviceFunc(
 			}
 			return deviceToDetach.GenerateErrorDetailed("GetDeviceMountRefs check failed", err)
 		}
-		// Execute unmount
-		unmountDeviceErr := volumeDetacher.UnmountDevice(deviceMountPath)
-		if unmountDeviceErr != nil {
-			// On failure, return error. Caller will log and retry.
-			return deviceToDetach.GenerateErrorDetailed("UnmountDevice failed", unmountDeviceErr)
-		}
-		// Before logging that UnmountDevice succeeded and moving on,
-		// use mounter.PathIsDevice to check if the path is a device,
-		// if so use mounter.DeviceOpened to check if the device is in use anywhere
-		// else on the system. Retry if it returns true.
-		isDevicePath, devicePathErr := mounter.PathIsDevice(deviceToDetach.DevicePath)
-		var deviceOpened bool
-		var deviceOpenedErr error
-		if !isDevicePath && devicePathErr == nil {
-			// not a device path or path doesn't exist
-			//TODO: refer to #36092
-			glog.V(3).Infof("Not checking device path %s", deviceToDetach.DevicePath)
-			deviceOpened = false
-		} else {
-			deviceOpened, deviceOpenedErr = mounter.DeviceOpened(deviceToDetach.DevicePath)
-			if deviceOpenedErr != nil {
-				return deviceToDetach.GenerateErrorDetailed("UnmountDevice.DeviceOpened failed", deviceOpenedErr)
+		// Try to unmap symlink on deviceMountPath
```

Comment:
Why are there changes to Unmount when we have a separate Unmap function?

Reply:
I need to check this again whether we can get persistent volume object from AttachedVolume struct.
If we can get PV object, then we can make this method separately.